lib/email-addresses.js

```diff
@@ -822,9 +853,23 @@ function parse5322(opts) {
                 delete ret[i].node;
             }
         }
-        return ret;
+        return ret;        
     }
-
+    
+    function addrListGiveResult(ast) {
+        return _giveResult(ast, ['group', 'mailbox']);
```

Comment:
Is it intentional that `addrListGiveResult`, `fromGiveResult`, and `senderGiveResult` have the same body of code? Is there a reason they may need to have different functionality? Either one should be different or there should be one function and no `giveResMap`. I do see that you wanted to hide the interface of `_giveResult`, but no need to have different function names.